libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -264,13 +272,13 @@ void AC_Avoid::adjust_velocity_proximity(float kP, float accel_cmss, Vector2f &d
     // get boundary from proximity sensor
     uint16_t num_points;
     const Vector2f *boundary = _proximity.get_boundary_points(num_points);
-    adjust_velocity_polygon(kP, accel_cmss, desired_vel, boundary, num_points, false);
+    adjust_velocity_polygon(kP, accel_cmss, desired_vel, boundary, num_points, false, _margin);
```

Comment:
fence.get_margin() gets the minimum distance the vehicle should maintain from the edge of the fence.  The _margin (which refers to the AVOID_MARGIN parameter) is the distance the vehicle should maintain from objects.   It's possible that the Fence margin should move into the Avoidance library but I think keeping it in the fence library (where it has been) is more intuitive for people during setup.
We could argue that the object avoidance margin should be moved out of the Avoidance library and into the proximity library but I think the placement is not terrible.  The Proximity library is really meant just to gather up the distances from whatever sensors.